envoy/common/execution_context.h

```diff
@@ -62,7 +66,18 @@ class ScopedExecutionContext : NonCopyable {
   bool isNull() const { return context_ == nullptr; }
 
 private:
+  ExecutionContext* getExecutionContext(OptRef<const StreamInfo::StreamInfo> info) {
+    if (!info.has_value()) {
+      return nullptr;
+    }
+    const auto* const_context = info->filterState().getDataReadOnly<ExecutionContext>(
+        kConnectionExecutionContextFilterStateName);
+    return const_cast<ExecutionContext*>(const_context);
+  }
+
   ExecutionContext* context_;
 };
 
+#endif
```

Comment:
Is the idea to add `ENVOY_EXECUTION_SCOPE` like below, in a future PR?

```
#ifdef ENVOY_ENABLE_EXECUTION_CONTEXT
#define ENVOY_EXECUTION_SCOPE(streamInfo, scopedObject) \
    getExecutionContext(streamInfo)->onScopeEnter(scopedObject)
#else 
#define ENVOY_EXECUTION_SCOPE as noop
#endif
```

(If yes, we may want to change `getExecutionContext` to a global function.)

Reply:
yeah.

Reply:
Feel free to submit as is. I can change `getExecutionContext` to a global function in my `ENVOY_EXECUTION_SCOPE` PR.